tests/rptest/tests/compatibility/sarama_test.py

```diff
@@ -0,0 +1,186 @@
+        time.sleep(10)
+        subprocess.run(["kill -2 $(ps -ef | grep interceptors | grep -v \"grep\" | awk '{print $2}')"], shell=True)
+        self.logger.debug("killed interceptors pipe")
+
+        p.wait()
+
+        event_found = False
+        for line in iter(p.stdout.readline, b''):
+            if b'SpanContext' in line.rstrip():
+                event_found = True
+
+        if not event_found:
+            raise RuntimeError("sarama interceptors test failed: {}".format(
+                event_found))
+
+    # Run sarama http server example
+    def test_sarama_http_server(self):
+        EXAMPLE_DIR = os.path.join(SaramaTest.TESTS_DIR, "examples/http_server")
+
+        #Tidy the example
+        self._execute("go mod tidy", EXAMPLE_DIR)
+
+        #Build the example
+        self._execute("go build", EXAMPLE_DIR)
+
+        #Run the example
+        cmd = "./http_server -addr localhost:8080 -brokers n1:9092"
+        p = subprocess.Popen([cmd],
+                        stdout=subprocess.PIPE,
+                        stderr=subprocess.STDOUT,
+                        shell=True,
+                        cwd=EXAMPLE_DIR)
+
+        # give http server time to load
+        time.sleep(2)
+
+        # requests on server
+        page = requests.get("http://localhost:8080/")
+        self.logger.debug(page.text)
+        success = page.ok
+
+        # kill server
+        subprocess.run(["kill -2 $(ps -ef | grep http_server | grep -v \"grep\" | awk '{print $2}')"], shell=True)
+        self.logger.debug("killed http_server pipe")
+
+        p.wait()
+        if not success:
+            raise RuntimeError("sarama http_server test failed: {}".format(
+                event_found))
+
+    # run sarama consumergroup example
+    @cluster(num_nodes=3)
+    def test_sarama_consumergroup(self):
+        EXAMPLE_DIR = os.path.join(SaramaTest.TESTS_DIR, "examples/consumergroup")
+
+        #Tidy the example
+        self._execute("go mod tidy", EXAMPLE_DIR)
+
+        #Build the example
+        self._execute("go build", EXAMPLE_DIR)
+
+        #Create and publish to sarama topic
+        message = 'redpanda is fast'
+        key = 'key'
+        h_key = 'h_key'
+        h_value = 'h_value'
+        headers = [h_key + ':' + h_value]
+        self._rpk.create_topic(self.sarama_topic)
+        self._rpk.produce(self.sarama_topic, key, message, headers)
+
+        #Run the example
+        cmd = "./consumer -brokers=\"n1:9092\" -topics=\"{}\" -group=\"example\"".format(self.sarama_topic)
+        p = subprocess.Popen([cmd],
+                        stdout=subprocess.PIPE,
+                        stderr=subprocess.STDOUT,
+                        shell=True,
+                        cwd=EXAMPLE_DIR)
+
+        # give example time to run
+        time.sleep(5)
+
+        # kill consumer
+        subprocess.run(["kill -2 $(ps -ef | grep consumer | grep -v \"grep\" | awk '{print $2}')"], shell=True)
```

Comment:
once the examples are built in the Dockerfile and available on all the ducktape nodes, you can wrap the the consumer/producer in a background threaded service. it will handle these sort of thing and make the tests much simpler and robust.

Reply:
That makes sense. I will make the change.